dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java

```diff
@@ -244,4 +297,633 @@ public final Response switchSite(
+    public Response makeDefault(@Context final HttpServletRequest httpServletRequest,
+                           @Context final HttpServletResponse httpServletResponse,
+                           @PathParam("hostId")  final String hostId) throws DotDataException, DotSecurityException, PortalException, SystemException {
+
+        final User user = new WebResource.InitBuilder(this.webResource)
+                .requestAndResponse(httpServletRequest, httpServletResponse)
+                .requiredBackendUser(true)
+                .rejectWhenNoUser(true)
+                .requireLicense(true)
+                .init().getUser();
+
+        Logger.debug(this, ()-> "making the site: " + hostId + " as a default");
+
+        final PageMode      pageMode      = PageMode.get(httpServletRequest);
+        final boolean respectFrontendRoles = pageMode.respectAnonPerms;
+        final Host host = pageMode.respectAnonPerms? this.siteHelper.getSite(user, hostId):
+                this.siteHelper.getSiteNoFrontEndRoles(user, hostId);
+
+        if (null == host) {
+            throw new IllegalArgumentException("Site: " + hostId + " does not exists");
+        }
+
+        return Response.ok(new ResponseEntityView(
+                this.siteHelper.makeDefault(host, user, respectFrontendRoles))).build();
+    }
+
+    /**
+     * Get the host setup progress when the hosts assets are being copied on background
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param hostId
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws PortalException
+     * @throws SystemException
+     */
+    @GET
+    @Path("/_setup_progress/{hostId}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response getSiteSetupProgress(@Context final HttpServletRequest httpServletRequest,
+                                @Context final HttpServletResponse httpServletResponse,
+                                @PathParam("hostId")  final String hostId) throws DotDataException, DotSecurityException, PortalException, SystemException {
+
+        Logger.debug(this, ()-> "Getting the site : " + hostId + " as a default");
+
+        return Response.ok(new ResponseEntityView(
+                QuartzUtils.getTaskProgress("setup-host-" + hostId, "setup-host-group"))).build();
+    }
+
+    /**
+     * Finds the host by identifier
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param hostId
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws PortalException
+     * @throws SystemException
+     */
+    @GET
+    @Path("/{hostId}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response findHostByIdentifier(@Context final HttpServletRequest httpServletRequest,
```

Comment:
Maybe we can merge this endpoint and the host by name endpoint into one, and pass the values to search by query param. `?identifier=xxxxxx&name=yyyy`

Also it's worth to be able to search by some part of the identifier or the name, e.g:
```
2 sites:
- sitenumberone
- sitenumbertwo

We should be able to search: `?name=sitenumber` and should return both site.
```

Behaviour could be something like the `TemplateResource#list`

Reply:
@erickgonzalez thing is @jdotcms is sending the name in the body to avoid escaping.

Reply:
yeah, well I have just migrated what the API was. Not implementing anything new here.
As @dsilvam said, both a similar but diff since we can not send the host name by query string b/c the escaping issues